mesh/pmesh.cpp

```diff
@@ -1776,6 +1797,28 @@ void ParMesh::SetCurvature(int order, bool discont, int space_dim, int ordering)
    Nodes->MakeOwner(nfec);
 }
 
+void ParMesh::EnsureParNodes()
```

Comment:
Is there any reason why we don't simply virtualize and overwrite `EnsureNodes`?

Reply:
I believe the two methods are fundamentally different. `Mesh::EnsureNodes` is a public method that makes sure for the user that `Nodes` is defined, even if the mesh has straight edges. On the other hand, `ParMesh::EnsureParNodes` is an internal helper that converts `Nodes` (if they exist) from a `GridFunction` to a `ParGridFunction`.

`EnsureParNodes` is actually a refactor of the messy block in `ParMesh::RebalanceImpl` in master. Maybe I should have chosen a less similar name to `EnsureNodes` (then again, the name implies what actually happens).

Yet another question is whether `Mesh::EnsureNodes` works correctly if called on a `ParMesh`, but this is out of the scope of this PR.

Reply:
@jakubcerveny, `ParMesh::SetCurvature` may already do what you need (it's not exactly the same as your `EnsureParNodes`, but it's close)